src/t_string.c

```diff
@@ -118,7 +117,7 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
         /* Propagate as SET Key Value PXAT millisecond-timestamp if there is
          * EX/PX/EXAT/PXAT flag. */
         robj *milliseconds_obj = createStringObjectFromLongLong(milliseconds);
-        rewriteClientCommandVector(c, 5, shared.set, key, val, shared.pxat, milliseconds_obj);
+        rewriteClientCommandVector(c, 5, shared.set, key, *val, shared.pxat, milliseconds_obj);
```

Comment:
I don't see any explicit interaction with https://github.com/redis/redis/issues/11356, but it's touching the same code path so raising it to see if other folks see issues.